src/MPoly.jl

```diff
@@ -425,16 +425,27 @@ end
 
 iszero(x::MPolyRingElem{T}) where T <: RingElement = length(x) == 0
 
-function is_unit(a::MPolyRingElem{T}) where T <: RingElement
-   if is_constant(a)
-      return is_unit(leading_coefficient(a))
-   elseif is_domain_type(elem_type(coefficient_ring(a)))
+# function is_unit(a::MPolyRingElem{T}) where T <: RingElement
+#    if is_constant(a)
+#       return is_unit(leading_coefficient(a))
+#    elseif is_domain_type(elem_type(coefficient_ring(a)))
+#       return false
+#    elseif length(a) == 1
+#       return false
+#    else
+#       throw(NotImplementedError(:is_unit, a))
+#    end
+# end
+function is_unit(f::T) where {T<:MPolyRingElem}
+  is_unit(constant_coefficient(f)) || return false
+  characteristic(parent(f)) == 1 && return true  # coeffs in zero ring
+  is_domain_type(elem_type(coefficient_ring(f))) && return (length(f)==1)
```

Comment:
Isn't this equivalent to this (which is almost the old code just with one redundant case dropped and formatted with `&&` instead of `if/else`):
```suggestion
  is_constant(f) && return is_unit(constant_coefficient(f))
  is_domain_type(elem_type(coefficient_ring(f))) && return false
```
In particular if the polynomial ring is trivial then `is_constant(f)` should return `true anyway. And `length(f)==1` is just yet another way to express "is constant (but not zero)". 

Of course we also could add some comments:
```suggestion
  # for constant polynomials we simply delegate to the base ring
  is_constant(f) && return is_unit(constant_coefficient(f))

  # over an integral domain, non-constant polynomials are never units
  is_domain_type(elem_type(coefficient_ring(f))) && return false
```

Reply:
Not quite equivalent: we must check that `is_unit(constant_coefficient(f))` before the `is_domain_type` check.  Luckily the test-suite detected the mistake... I didn't spot it (oops)

Reply:
Seems to be OK now... just waiting for GitHub to confirm.  Running the tests locally went OK.